genesyscloud/resource_genesyscloud_knowledge_document_test.go

```diff
@@ -0,0 +1,212 @@
+        }
+        `, varType,
+		externalUrl,
+		generateFaq(question, answer, faqAlternatives),
+		knowledgeCategoryResourceName)
+}
+
+func generateKnowledgeDocumentRequestBodyArticle(varType string, externalUrl string, title string, contentLocationUrl string, alternatives []string, knowledgeCategoryResourceName []string) string {
+	return fmt.Sprintf(`
+        knowledge_document {
+            type = "%s"
+            external_url = "%s"
+            %s
+            categories = [genesyscloud_knowledge_category.%s.knowledge_category.0.name]
+        }
+        `, varType,
+		externalUrl,
+		generateArticle(title, contentLocationUrl, alternatives),
+		knowledgeCategoryResourceName)
+}
+
+func generateFaq(question string, answer string, alternatives []string) string {
+	formattedAlternatives := ""
+	for i, alternative := range alternatives {
+		if i > 0 {
+			formattedAlternatives += ", "
+		}
+		formattedAlternatives += strconv.Quote(alternative)
+	}
+
+	return fmt.Sprintf(`
+        faq {
+            question = "%s"
+            answer = "%s"
+            alternatives = [%s]
+        }
+        `, question,
+		answer,
+		formattedAlternatives)
+}
+
+func generateArticle(title string, contentLocationUrl string, alternatives []string) string {
+	formattedAlternatives := ""
+	for i, alternative := range alternatives {
+		if i > 0 {
+			formattedAlternatives += ", "
+		}
+		formattedAlternatives += strconv.Quote(alternative)
+	}
+
+	return fmt.Sprintf(`
+        article {
+            title = "%s"
+            content_location_url = "%s"
+            alternatives = [%s]
+        }
+        `, title,
+		contentLocationUrl,
+		formattedAlternatives)
+}
+
+func testVerifyKnowledgeDocumentDestroyed(state *terraform.State) error {
+	knowledgeAPI := platformclientv2.NewKnowledgeApi()
+	knowledgeBaseCoreLanguage1 := "en-US"
+	var knowledgeBaseId string
+	for _, rs := range state.RootModule().Resources {
+		if rs.Type == "genesyscloud_knowledge_knowledgebase" {
+			knowledgeBaseId = rs.Primary.ID
+			break
+		}
+	}
+	for _, rs := range state.RootModule().Resources {
+		if rs.Type != "genesyscloud_knowledge_document" {
+			continue
+		}
+		id := strings.Split(rs.Primary.ID, " ")
+		knowledgeDocumentId := id[0]
+		knowledgeDocument, resp, err := knowledgeAPI.GetKnowledgeKnowledgebaseLanguageDocument(knowledgeDocumentId, knowledgeBaseId, knowledgeBaseCoreLanguage1)
+		if knowledgeDocument != nil {
+			return fmt.Errorf("Knowledge document (%s) still exists", knowledgeDocumentId)
+		} else if isStatus404(resp) || isStatus400(resp) {
+			// Knowledge base not found as expected
+			continue
+		} else {
+			// Unexpected error
+			return fmt.Errorf("Unexpected error: %s", err)
+		}
+	}
+	// Success. All knowledge bases destroyed
+	return nil
+}
+
+func generateKnowledgeDocumentLifeCycle() string {
+	return `
+	lifecycle {
+		ignore_changes = [
+			"knowledge_document[0].categories",           
```

Comment:
Why are we ignoring this change?

Reply:
I can remove this actually. I thought it was needed at some point but the test is passing without it now.